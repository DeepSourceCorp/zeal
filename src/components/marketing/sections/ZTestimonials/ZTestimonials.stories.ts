import ZTestimonials from './ZTestimonials.vue'
import ZTestimonial from '../../ZTestimonial'

const testimonials = [
  {
    customer: 'Dgraph',
    text:
      'DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.',
    author: 'Piero Molino, Senior Technology Officer',
    image: { color: 'https://i.imgur.com/ISsHpNY.png', bnw: 'https://i.imgur.com/ISsHpNY.png' },
    url: ''
  },
  {
    customer: 'Intel',
    text:
      'DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.',
    author: 'Piero Molino, Senior Technology Officer',
    image: { color: 'https://i.imgur.com/PYEc7bZ.png', bnw: 'https://i.imgur.com/PYEc7bZ.png' },
    url: ''
  },
  {
    customer: 'Qubole',
    text:
      'DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.',
    author: 'Piero Molino, Senior Technology Officer',
    image: { color: 'https://i.imgur.com/GQjbpeq.png', bnw: 'https://i.imgur.com/GQjbpeq.png' },
    url: ''
  },
  {
    customer: 'Okteto',
    text:
      'DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.',
    author: 'Piero Molino, Senior Technology Officer',
    image: { color: 'https://i.imgur.com/oqlV3mB.png', bnw: 'https://i.imgur.com/oqlV3mB.png' },
    url: ''
  },
  {
    customer: 'Slack1',
    text:
      'DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.',
    author: 'Piero Molino, Senior Technology Officer',
    image: { color: 'https://i.imgur.com/ifq7pvm.png', bnw: 'https://i.imgur.com/ifq7pvm.png' },
    url: ''
  },
  {
    customer: 'Slack2',
    text:
      'DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.',
    author: 'Piero Molino, Senior Technology Officer',
    image: { color: 'https://i.imgur.com/ifq7pvm.png', bnw: 'https://i.imgur.com/ifq7pvm.png' },
    url: ''
  }
]

export default {
  title: 'Zeal Marketing/Testimonials Section',
  component: ZTestimonials,
  excludeStories: /.*Data$/
}

export const Basic = () => ({
  components: { ZTestimonials, ZTestimonial },
  data() {
    return {
      testimonials
    }
  },
  template: `<div class='py-32 px-20'>
      <z-testimonials :testimonials="testimonials" :timing="10000">
        <h2 slot="heading">What developers say</h2>
      </z-testimonials>
    </div>`
})
