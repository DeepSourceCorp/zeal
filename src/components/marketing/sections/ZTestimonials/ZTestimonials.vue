<template>
  <section class="flex">
    <div class="flex flex-col justify-around relative">
      <span class="text-vanilla-100 font-bold text-4xl leading-12 mr-20">
        <slot name="heading"></slot>
      </span>
      <span class="grid grid-cols-3 mt-6 mb-4">
        <span
          v-for="(testimonial, index) in testimonials"
          :key="testimonial.customer"
          class="flex items-center mt-12 cursor-pointer"
          @click="setCurrentIndexTo(index)"
        >
          <img
            :class="[`${index === currentIndex ? '' : INACTIVE_TESTIMONIAL}`]"
            class="object-contain"
            :src="testimonial.image.bnw"
            :alt="testimonial.customer"
          />
        </span>
      </span>
    </div>
    <div class="w-4/6">
      <z-testimonial
        v-for="(testimonial, index) in testimonials"
        :key="testimonial.customer"
        class="bg-ink-300"
        :class="[`${index === currentIndex ? VISIBLE_CARD : INVISIBLE_CARD}`]"
      >
        <template slot="image">
          <img
            class="object-contain h-12"
            :src="testimonial.image.color"
            :alt="testimonial.customer"
          />
        </template>
        <template slot="body">
          {{ testimonial.text }}
        </template>
        <template slot="footer">
          {{ testimonial.author }}
        </template>
      </z-testimonial>
    </div>
  </section>
</template>

<script lang="ts">
import Vue from 'vue'
import ZTestimonial from '@/components/marketing/ZTestimonial'

const VISIBLE_CARD = 'block animate-fadeIn'
const INVISIBLE_CARD = 'hidden'
const INACTIVE_TESTIMONIAL = 'opacity-60'

const MAX_TESTIMONIALS = 6
const DEFAULT_INTERVAL = 5000

export default Vue.extend({
  name: 'ZTestimonials',
  components: {
    ZTestimonial
  },
  props: {
    testimonials: {
      required: true,
      type: Array,
      validator: (arr) => arr.length <= MAX_TESTIMONIALS
    },
    timing: {
      default: DEFAULT_INTERVAL,
      type: Number
    }
  },
  data() {
    return {
      currentIndex: 0,
      interval: 0,
      VISIBLE_CARD,
      INVISIBLE_CARD,
      INACTIVE_TESTIMONIAL
    }
  },
  mounted() {
    this.setTestimonialInterval()
  },
  methods: {
    setTestimonialInterval() {
      this.interval = setInterval(() => {
        this.showNextTestimonial()
      }, this.timing)
    },
    resetInterval() {
      clearInterval(this.interval)
      this.setTestimonialInterval()
    },
    showNextTestimonial() {
      this.currentIndex++
      if (this.currentIndex >= this.testimonials.length) {
        this.currentIndex = 0
      }
    },
    setCurrentIndexTo(index: number) {
      this.currentIndex = index
      this.resetInterval()
    }
  }
})
</script>
