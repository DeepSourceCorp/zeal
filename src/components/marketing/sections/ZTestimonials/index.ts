import ZTestimonials from './ZTestimonials.vue'

export default ZTestimonials
