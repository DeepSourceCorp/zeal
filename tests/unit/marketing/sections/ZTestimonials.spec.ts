import Vue from 'vue'
import { createLocalVue, mount, Wrapper } from '@vue/test-utils'

import ZTestimonials from '../../../../src/components/marketing/sections/ZTestimonials'

const testimonials = [
  {
    customer: 'Dgraph',
    text:
      'DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.',
    author: 'Piero Molino, Senior Technology Officer',
    image: { color: 'https://i.imgur.com/ISsHpNY.png', bnw: 'https://i.imgur.com/ISsHpNY.png' },
    url: ''
  },
  {
    customer: 'Intel',
    text:
      'DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.',
    author: 'Piero Molino, Senior Technology Officer',
    image: { color: 'https://i.imgur.com/oqlV3mB.png', bnw: 'https://i.imgur.com/oqlV3mB.png' },
    url: ''
  },
  {
    customer: 'Qubole',
    text:
      'DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.',
    author: 'Piero Molino, Senior Technology Officer',
    image: { color: 'https://i.imgur.com/ISsHpNY.png', bnw: 'https://i.imgur.com/ISsHpNY.png' },
    url: ''
  },
  {
    customer: 'Okteto',
    text:
      'DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.',
    author: 'Piero Molino, Senior Technology Officer',
    image: { color: 'https://i.imgur.com/oqlV3mB.png', bnw: 'https://i.imgur.com/oqlV3mB.png' },
    url: ''
  },
  {
    customer: 'Slack1',
    text:
      'DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.',
    author: 'Piero Molino, Senior Technology Officer',
    image: { color: 'https://i.imgur.com/ifq7pvm.png', bnw: 'https://i.imgur.com/ifq7pvm.png' },
    url: ''
  },
  {
    customer: 'Slack2',
    text:
      'DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.',
    author: 'Piero Molino, Senior Technology Officer',
    image: { color: 'https://i.imgur.com/ifq7pvm.png', bnw: 'https://i.imgur.com/ifq7pvm.png' },
    url: ''
  }
]

describe('Testimonials section', () => {
  let mountFunction: (options?: Record<string, unknown>) => Wrapper<Vue>
  let localVue: typeof Vue
  beforeEach(() => {
    localVue = createLocalVue()
    mountFunction = (options = {}) => {
      return mount(ZTestimonials, {
        localVue,
        ...options
      })
    }
  })
  it('renders testimonials section when the component is mounted', () => {
    let mountedInstance = mountFunction({
      propsData: {
        testimonials
      }
    })
    expect(mountedInstance.html()).toMatchSnapshot()
  })

  it('renders testimonials section heading when given in slot', () => {
    let mountedInstance = mountFunction({
      propsData: {
        testimonials
      },
      slots: {
        heading: 'This is heading'
      }
    })
    expect(mountedInstance.text()).toContain('This is heading')
  })

  it('renders six testimonial logos but renders one testimonial card', () => {
    let mountedInstance = mountFunction({
      propsData: {
        testimonials
      }
    })
    expect(mountedInstance.findAll('.z-testimonials__logo').length).toEqual(6)
    expect(mountedInstance.findAll('.bg-ink-300').length).toEqual(6)
    expect(mountedInstance.findAll('.block.animate-fadeIn').length).toEqual(1)
  })
})
