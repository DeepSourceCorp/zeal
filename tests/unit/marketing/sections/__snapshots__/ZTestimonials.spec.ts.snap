// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonials section renders testimonials section when the component is mounted 1`] = `
<section class="lg:flex">
  <div class="flex flex-col justify-around relative"><span class="text-vanilla-100 font-bold text-3xl md:text-4xl leading-11 lg:leading-12 md:mr-20"></span> <span class="z-testimonials__logos flex lg:grid grid-cols-3 px-8 sm:px-0 mt-6 mb-2 overflow-x-hidden flex-nowrap w-full-screen lg:w-full-screen-none"><span class="z-testimonials__logo flex-shrink-0 flex items-center mt-8 lg:mt-12 cursor-pointer mr-8 w-24 md:w-auto"><img src="https://i.imgur.com/ISsHpNY.png" alt="Dgraph" class="object-contain hover:opacity-100"></span><span class="z-testimonials__logo flex-shrink-0 flex items-center mt-8 lg:mt-12 cursor-pointer mr-8 w-24 md:w-auto"><img src="https://i.imgur.com/oqlV3mB.png" alt="Intel" class="object-contain hover:opacity-100 opacity-60"></span><span class="z-testimonials__logo flex-shrink-0 flex items-center mt-8 lg:mt-12 cursor-pointer mr-8 w-24 md:w-auto"><img src="https://i.imgur.com/ISsHpNY.png" alt="Qubole" class="object-contain hover:opacity-100 opacity-60"></span><span class="z-testimonials__logo flex-shrink-0 flex items-center mt-8 lg:mt-12 cursor-pointer mr-8 w-24 md:w-auto"><img src="https://i.imgur.com/oqlV3mB.png" alt="Okteto" class="object-contain hover:opacity-100 opacity-60"></span><span class="z-testimonials__logo flex-shrink-0 flex items-center mt-8 lg:mt-12 cursor-pointer mr-8 w-24 md:w-auto"><img src="https://i.imgur.com/ifq7pvm.png" alt="Slack1" class="object-contain hover:opacity-100 opacity-60"></span><span class="z-testimonials__logo flex-shrink-0 flex items-center mt-8 lg:mt-12 cursor-pointer mr-8 w-24 md:w-auto"><img src="https://i.imgur.com/ifq7pvm.png" alt="Slack2" class="object-contain hover:opacity-100 opacity-60"></span></span> <span class="flex justify-between -mt-16 w-full-screen lg:w-full-screen-none lg:hidden pointer-events-none"><span class="w-20 bg-gradient-to-r from-ink-400 via-transparent to-transparent h-16"></span> <span class="w-40 bg-gradient-to-r from-transparent via-transparent to-ink-400 h-16"></span></span></div>
  <div class="lg:w-4/6">
    <div class="rounded-sm md:rounded-md overflow-hidden transition-all shadow flex flex-col justify-around h-full px-5 py-4 md:px-10 md:py-8 bg-ink-300 my-4 lg:my-0 block animate-fadeIn">
      <div class="flex-1"><img src="https://i.imgur.com/ISsHpNY.png" alt="Dgraph" class="object-contain h-12 hidden lg:block mb-6"></div> <span class="flex-1"><div class="text-vanilla-100 text-md md:text-xl mb-6 pr-2">
        DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.
      </div> <div class="text-vanilla-400 text-md md:text-lg">
        Piero Molino, Senior Technology Officer
      </div></span>
    </div>
    <div class="rounded-sm md:rounded-md overflow-hidden transition-all shadow flex flex-col justify-around h-full px-5 py-4 md:px-10 md:py-8 bg-ink-300 my-4 lg:my-0 hidden">
      <div class="flex-1"><img src="https://i.imgur.com/oqlV3mB.png" alt="Intel" class="object-contain h-12 hidden lg:block mb-6"></div> <span class="flex-1"><div class="text-vanilla-100 text-md md:text-xl mb-6 pr-2">
        DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.
      </div> <div class="text-vanilla-400 text-md md:text-lg">
        Piero Molino, Senior Technology Officer
      </div></span>
    </div>
    <div class="rounded-sm md:rounded-md overflow-hidden transition-all shadow flex flex-col justify-around h-full px-5 py-4 md:px-10 md:py-8 bg-ink-300 my-4 lg:my-0 hidden">
      <div class="flex-1"><img src="https://i.imgur.com/ISsHpNY.png" alt="Qubole" class="object-contain h-12 hidden lg:block mb-6"></div> <span class="flex-1"><div class="text-vanilla-100 text-md md:text-xl mb-6 pr-2">
        DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.
      </div> <div class="text-vanilla-400 text-md md:text-lg">
        Piero Molino, Senior Technology Officer
      </div></span>
    </div>
    <div class="rounded-sm md:rounded-md overflow-hidden transition-all shadow flex flex-col justify-around h-full px-5 py-4 md:px-10 md:py-8 bg-ink-300 my-4 lg:my-0 hidden">
      <div class="flex-1"><img src="https://i.imgur.com/oqlV3mB.png" alt="Okteto" class="object-contain h-12 hidden lg:block mb-6"></div> <span class="flex-1"><div class="text-vanilla-100 text-md md:text-xl mb-6 pr-2">
        DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.
      </div> <div class="text-vanilla-400 text-md md:text-lg">
        Piero Molino, Senior Technology Officer
      </div></span>
    </div>
    <div class="rounded-sm md:rounded-md overflow-hidden transition-all shadow flex flex-col justify-around h-full px-5 py-4 md:px-10 md:py-8 bg-ink-300 my-4 lg:my-0 hidden">
      <div class="flex-1"><img src="https://i.imgur.com/ifq7pvm.png" alt="Slack1" class="object-contain h-12 hidden lg:block mb-6"></div> <span class="flex-1"><div class="text-vanilla-100 text-md md:text-xl mb-6 pr-2">
        DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.
      </div> <div class="text-vanilla-400 text-md md:text-lg">
        Piero Molino, Senior Technology Officer
      </div></span>
    </div>
    <div class="rounded-sm md:rounded-md overflow-hidden transition-all shadow flex flex-col justify-around h-full px-5 py-4 md:px-10 md:py-8 bg-ink-300 my-4 lg:my-0 hidden">
      <div class="flex-1"><img src="https://i.imgur.com/ifq7pvm.png" alt="Slack2" class="object-contain h-12 hidden lg:block mb-6"></div> <span class="flex-1"><div class="text-vanilla-100 text-md md:text-xl mb-6 pr-2">
        DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.
      </div> <div class="text-vanilla-400 text-md md:text-lg">
        Piero Molino, Senior Technology Officer
      </div></span>
    </div>
    <ul class="flex mx-auto mt-4 justify-between lg:hidden"><span class="cursor-pointer"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="stroke-1.5 z-icon w-4 h-4 sm:w-6 sm:h-6 z-icon--arrow-left text-slate"><line x1="19" y1="12" x2="5" y2="12"></line><polyline points="12 19 5 12 12 5"></polyline></svg></span> <span class="flex items-center"><li class="h-2 w-2 cursor-pointer rounded-full mx-2 bg-juniper"><button></button></li><li class="h-2 w-2 cursor-pointer rounded-full mx-2 bg-ink-100"><button></button></li><li class="h-2 w-2 cursor-pointer rounded-full mx-2 bg-ink-100"><button></button></li><li class="h-2 w-2 cursor-pointer rounded-full mx-2 bg-ink-100"><button></button></li><li class="h-2 w-2 cursor-pointer rounded-full mx-2 bg-ink-100"><button></button></li><li class="h-2 w-2 cursor-pointer rounded-full mx-2 bg-ink-100"><button></button></li></span> <span class="cursor-pointer"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="stroke-1.5 z-icon w-4 h-4 sm:w-6 sm:h-6 z-icon--arrow-right text-slate"><line x1="5" y1="12" x2="19" y2="12"></line><polyline points="12 5 19 12 12 19"></polyline></svg></span></ul>
  </div>
</section>
`;
